File: python/sorting/Quick_Sort.py
def partition(data, start, end):
    i = start + 1
    pivot = data[start]

    for j in range(start + 1, end + 1):
        if data[j] < pivot:
            data[i], data[j] = data[j], data[i]
            i += 1
    data[start], data[i - 1] = data[i - 1], data[start]
    return i - 1

def Quick_Sort(data, start, end):
    if start < end:
        pivot_position = partition(data, start, end)
        Quick_Sort(data, start, pivot_position - 1)
        Quick_Sort(data, pivot_position + 1, end)

File: java/sorting/Quick_Sort.java
package Algorithms_Code.Java;

public class Quick_Sort {
 
    int partition (int a[], int start, int end)  {  
        int pivot = a[end];  
        int i = (start - 1);  
        for (int j = start; j <= end - 1; j++)  {  
            if (a[j] < pivot){  
                i++;  
                int t = a[i];  
                a[i] = a[j];  
                a[j] = t;  
            }  
        }  
        int t = a[i + 1];  
        a[i + 1] = a[end];  
        a[end] = t;  
        return (i + 1);  
    }  
        
    void quick_sort(int a[], int start, int end){  
        if (start < end)  {  
            int p = partition(a, start, end);  
            quick_sort(a, start, p - 1);  
            quick_sort(a, p + 1, end);  
        }
    }  

    public static void main(String args[]){
        
    }

}
